06-TimeManagement-ModuleAbsoluteTime/absolute-time.c

```diff
@@ -0,0 +1,67 @@
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/device.h>
+#include <linux/err.h>
+#include <linux/uaccess.h>
+#include <linux/time.h>
+#include <linux/types.h>
+
+struct timespec ts;
+
+
+static ssize_t ro_show(struct class *class,
+		struct class_attribute *attr,
+		char *buf)
+{
+	char text[100];
+
+	sprintf(text, "%ld.%06ld\n", ts.tv_sec, ts.tv_nsec);
```

Comment:
Is this format string correct for values in ts structure?

Reply:
yes. it is.

Reply:
`$ make
make -C /home/the4th/Projects/GL/linux_build/ M=/home/the4th/Projects/GL/myFork/gl-kernel-training-2019.git/06-TimeManagement-ModuleAbsoluteTime modules
make[1]: Entering directory '/home/the4th/Projects/GL/linux_build'
  CC [M]  /home/the4th/Projects/GL/myFork/gl-kernel-training-2019.git/06-TimeManagement-ModuleAbsoluteTime/absolute-time.o
/home/the4th/Projects/GL/myFork/gl-kernel-training-2019.git/06-TimeManagement-ModuleAbsoluteTime/absolute-time.c: In function ‘ro_show’:
/home/the4th/Projects/GL/myFork/gl-kernel-training-2019.git/06-TimeManagement-ModuleAbsoluteTime/absolute-time.c:18:18: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘__kernel_time_t {aka long int}’ [-Wformat=]
  sprintf(text, "%d.%06ld\n", ts.tv_sec, ts.tv_nsec);
                 ~^           ~~~~~~~~~
                 %ld
/home/the4th/Projects/GL/myFork/gl-kernel-training-2019.git/06-TimeManagement-ModuleAbsoluteTime/absolute-time.c: In function ‘mymodule_init’:
/home/the4th/Projects/GL/myFork/gl-kernel-training-2019.git/06-TimeManagement-ModuleAbsoluteTime/absolute-time.c:39:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  int ret = class_create_file(attr_class, &class_attr_ro);
  ^~~
  Building modules, stage 2.
  MODPOST 1 modules
  CC      /home/the4th/Projects/GL/myFork/gl-kernel-training-2019.git/06-TimeManagement-ModuleAbsoluteTime/absolute-time.mod.o
  LD [M]  /home/the4th/Projects/GL/myFork/gl-kernel-training-2019.git/06-TimeManagement-ModuleAbsoluteTime/absolute-time.ko
make[1]: Leaving directory '/home/the4th/Projects/GL/linux_build'
`

Reply:
yes. It is.

Reply:
/home/the4th/Projects/GL/myFork/gl-kernel-training-2019.git/06-TimeManagement-ModuleAbsoluteTime/absolute-time.c:18:18: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘__kernel_time_t {aka long int}’ [-Wformat=]
  sprintf(text, "%d.%06ld\n", ts.tv_sec, ts.tv_nsec);
                 ~^           ~~~~~~~~~
                 %ld

Reply:
tv_sec is ok. What about %06ld  and tv_nsec?

Reply:
ok. I get it)